File: backend/user-service/src/main/java/it/roadies/user_service/data/dto/response/UserResponseDTO.java
package it.roadies.user_service.data.dto.response;

import it.roadies.user_service.data.entities.enumeration.Badge;
import lombok.Data;

@Data
public class UserResponseDTO {
    private String keycloakId;
    private String username;
    private String firstName;
    private String lastName;
    private String email;
    private String avatarUrl;

    private Long points;
    private Badge badge;

    private boolean enabled;
}

File: android-app/app/src/main/java/it/roadies/android_app/client/models/user/UserResponseDTO.kt
package it.roadies.android_app.client.models.user

import com.google.gson.annotations.SerializedName

data class UserResponseDTO(
    @SerializedName("keycloakId") val keycloakId: String,
    @SerializedName("username") val username: String?,
    @SerializedName("firstName") val firstName: String?,
    @SerializedName("lastName") val lastName: String?,
    @SerializedName("email") val email: String?,
    @SerializedName("avatarUrl") val avatarUrl: String?,
    @SerializedName("points") val points: Long?,
    @SerializedName("badge") val badge: String?,
    @SerializedName("enabled") val enabled: Boolean?
)